src/lidar_visualizer/visualizer.py

```diff
@@ -103,86 +111,142 @@ def _get_current_filename(self, idx):
+        points, colors = self._get_frame(self.idx)
+        self._register_frame(points, colors)
+
+    def _register_frame(self, points, colors):
+        frame_cloud = self._ps.register_point_cloud(
+            "current_frame",
+            points,
+            point_render_mode="quad",
+        )
+        if colors is None:
+            frame_cloud.set_color(FRAME_COLOR)
+        else:
+            frame_cloud.add_color_quantity("colors", colors, enabled=True)
+        frame_cloud.set_radius(self._frame_size, relative=False)
+        frame_cloud.set_enabled(self._toggle_frame)
+
+    def _compute_speed(self):
+        return 0.1 - 0.02 * self._speed_level
+
+    # GUI Callbacks ---------------------------------------------------------------------------
+    def _main_gui_callback(self):
+        self._gui.TextUnformatted("Controls:")
+        if not self.end_reached:
+            self._start_pause_callback()
+            if not self._play_mode:
+                self._gui.SameLine()
+                self._next_frame_callback()
+                if self._random_accessible_dataset:
+                    self._gui.SameLine()
+                    self._previous_frame_callback()
+        self._gui.Separator()
+        self._progress_bar_callback()
+        self._speed_level_callback()
+        self._gui.Separator()
+        self._gui.TextUnformatted("Scene Options:")
+        self._background_color_callback()
+        self._points_controlles_callback()
+        if not self._random_accessible_dataset:
+            self._gui.Separator()
+            self._information_callback()
+        self._gui.Separator()
+        self._center_viewpoint_callback()
+        self._gui.SameLine()
+        self._quit_callback()
+
+    def _start_pause_callback(self):
+        button_name = PAUSE_BUTTON if self._play_mode else START_BUTTON
+        if self._gui.Button(button_name) or self._gui.IsKeyPressed(self._gui.ImGuiKey_Space):
+            self._play_mode = not self._play_mode
+
+    def _next_frame_callback(self):
+        if self._gui.Button(NEXT_FRAME_BUTTON) or self._gui.IsKeyPressed(self._gui.ImGuiKey_N):
+            self.advance()
+            self._update_visualized_frame()
+
+    def _previous_frame_callback(self):
+        if self._gui.Button(PREVIOUS_FRAME_BUTTON) or self._gui.IsKeyPressed(self._gui.ImGuiKey_P):
+            self.rewind()
+            self._update_visualized_frame()
+
+    def _progress_bar_callback(self):
+        changed, idx = self._gui.SliderInt(
+            f"\t{self.stop_idx} Frames###Progress Bar",
+            self.idx,
+            v_min=self.start_idx,
+            v_max=self.stop_idx - 1,
+            format="Frame: %d",
+        )
+        if changed and self._random_accessible_dataset:
+            self.idx = idx
+            self._update_visualized_frame()
+
+    def _speed_level_callback(self):
+        _, self._speed_level = self._gui.SliderInt(
+            "\tPlayback Speed",
+            self._speed_level,
```

Comment:
I would change the range of speed level to something more meaningful, such as playback delay, or FPS, etc.
At the moment, having a Speed of `0` does not make sense